parser/src/parser/tests/mod.rs

```diff
@@ -12,6 +12,7 @@ mod arithmetic_ops;
 mod boundary_constraints;
 mod comments;
 mod constants;
+mod constraint_comprehension;
```

Comment:
I would probably prefer to put the tests in with the existing constraints tests, especially since it's handled differently for boundary vs. integrity constraints. We could use a section separator marking them as being related to comprehension